src/logic/player.h

```diff
@@ -700,7 +700,22 @@ class Player {
 
 	PlayerBuildingStats building_stats_;
 
-	std::map<std::tuple<unsigned, unsigned, unsigned, unsigned>, uint32_t> soldier_stats_;
+	struct SoldierStatistics {
+		const unsigned h, a, d, e;
```

Comment:
Please call these by their full name to make the code easier to read for people who are new to the codebase.